linux-tools/libnl/libnl.sh

```diff
@@ -170,17 +170,10 @@ function test_route_get()
 
 	nl-route-get $dst >$stdout 2>$stderr
 	tc_fail_if_bad $? "Execution of nl-route-get failed" || return
-
-	# We should see at least one line that starts with our destination address
-	# and the device which holds the route.
-	# $dst dev <dev> via <ip> scope <scope> *
-	grep "^$dst dev [a-zA-Z0-9\.]\+ via [0-9A-Fa-f:\.]\+ scope [a-zA-Z]\+" $stdout &>/dev/null
-
-	tc_pass_or_fail $? "Expected a route info: of the following fmt in o/p" \
-	"$dst dev <device> via <ip> scope <scope-name> *" \
-	"========= stdout ==============" \
-	"$(< $stdout)" \
-	"==============================" 
+        default_route=`grep preferred-src $stdout`
+        rout_ip=`echo $default_route |grep -o '[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}'`
+        ip route show | grep -q $rout_ip &>/dev/null
+        tc_pass_or_fail $? "Failed to get the route to an address in the network "    
 }
```

Comment:
From the output of nl-route-get:

# nl-route-get 10.0.0.1
inet cache 10.0.0.1 type unicast
    preferred-src <ip> scope global priority 0 protocol unspec
    nexthop via <route_ip> dev <device> <>

We should be verifying: <device> and <route_ip> 
In the patch code, preferred-src is being checked. 
Can you please check and verify